src/intersection-observer.js

```diff
@@ -218,10 +223,18 @@ export class IntersectionObserver extends Observable {
   }
 
   /**
-   * Provice a function to clear timeout before set this intersection to null.
+   * Provide a function to clear timeout before set this intersection to null.
    */
   destroy() {
     timer.cancel(this.flushTimeout_);
```

Comment:
I assume there's no need, because in every scenario where we call `destory()` we would have already set the baseElement `viewportCallback(false)`, and the intersectionObserver `onViewportCallback(false)` would unlisten them for us. 
https://github.com/ampproject/amphtml/blob/master/src/intersection-observer.js#L176